From 5d1a96488fe894d8d86b21cc9fb088da06bc6dd1 Mon Sep 17 00:00:00 2001
From: Andreas Dumberger <andreas.dumberger@tq-group.com>
Date: Wed, 7 Apr 2021 13:58:21 +0200
Subject: [PATCH] tqmarzg2x: set all GPIOs to input without PU/PD

The only exception are USER_LED_x pins, which are all set to output, high.
All GPIOs have a PU/PD as needed on the MB to ensure proper operation,
so we can safely remove these settings here.

Signed-off-by: Andreas Dumberger <andreas.dumberger@tq-group.com>
---
 drivers/renesas/rzg/pfc/G2H/pfc_init_h3_v2.c | 55 ++++++++++----------
 drivers/renesas/rzg/pfc/G2M/pfc_init_m3.c    | 55 ++++++++++----------
 drivers/renesas/rzg/pfc/G2N/pfc_init_m3n.c   | 55 ++++++++++----------
 3 files changed, 81 insertions(+), 84 deletions(-)

diff --git a/drivers/renesas/rzg/pfc/G2H/pfc_init_h3_v2.c b/drivers/renesas/rzg/pfc/G2H/pfc_init_h3_v2.c
index 24a414b7a..83bbedd2c 100644
--- a/drivers/renesas/rzg/pfc/G2H/pfc_init_h3_v2.c
+++ b/drivers/renesas/rzg/pfc/G2H/pfc_init_h3_v2.c
@@ -882,7 +882,6 @@ void pfc_init_h3_v2(void)
 	pfc_reg_write(PFC_GPSR6, GPSR6_AUDIO_CLKB_B
 		      | GPSR6_AUDIO_CLKA_A
 		      | GPSR6_SSI_WS6
-		      | GPSR6_SSI_SCK6
 		      | GPSR6_SSI_SDATA4
 		      | GPSR6_SSI_WS4
 		      | GPSR6_SSI_SCK4
@@ -1149,22 +1148,22 @@ void pfc_init_h3_v2(void)
 	pfc_reg_write(PFC_DRVCTRL24, reg);
 
 	/* initialize LSI pin pull-up/down control */
-	pfc_reg_write(PFC_PUD0,0x00005FBFU);
-	pfc_reg_write(PFC_PUD1,0x00300EFEU);
-	pfc_reg_write(PFC_PUD2,0x330001E6U);
-	pfc_reg_write(PFC_PUD3,0x000002E0U);
-	pfc_reg_write(PFC_PUD4,0xFFFFFF00U);
-	pfc_reg_write(PFC_PUD5,0x7F5FFF87U);
-	pfc_reg_write(PFC_PUD6,0x00000055U);
+	pfc_reg_write(PFC_PUD0,0x00000000U);
+	pfc_reg_write(PFC_PUD1,0x00000000U);
+	pfc_reg_write(PFC_PUD2,0x00000000U);
+	pfc_reg_write(PFC_PUD3,0x00000000U);
+	pfc_reg_write(PFC_PUD4,0x00000000U);
+	pfc_reg_write(PFC_PUD5,0x00000000U);
+	pfc_reg_write(PFC_PUD6,0x00000000U);
 
 	/* initialize LSI pin pull-enable register */
-	pfc_reg_write(PFC_PUEN0,0x00000FFFU);
-	pfc_reg_write(PFC_PUEN1,0x00100234U);
-	pfc_reg_write(PFC_PUEN2,0x000004C4U);
-	pfc_reg_write(PFC_PUEN3,0x00000200U);
-	pfc_reg_write(PFC_PUEN4,0x3E000000U);
-	pfc_reg_write(PFC_PUEN5,0x1F000805U);
-	pfc_reg_write(PFC_PUEN6,0x00000006U);
+	pfc_reg_write(PFC_PUEN0,0x00000000U);
+	pfc_reg_write(PFC_PUEN1,0x00000000U);
+	pfc_reg_write(PFC_PUEN2,0x00000000U);
+	pfc_reg_write(PFC_PUEN3,0x00000000U);
+	pfc_reg_write(PFC_PUEN4,0x00000000U);
+	pfc_reg_write(PFC_PUEN5,0x00000000U);
+	pfc_reg_write(PFC_PUEN6,0x00000000U);
 
 	/* initialize positive/negative logic select */
 	mmio_write_32(GPIO_POSNEG0, 0x00000000U);
@@ -1187,26 +1186,26 @@ void pfc_init_h3_v2(void)
 	mmio_write_32(GPIO_IOINTSEL7, 0x00000000U);
 
 	/* initialize general output register */
-	mmio_write_32(GPIO_OUTDT0, 0x00000001U);
+	mmio_write_32(GPIO_OUTDT0, 0x00000000U);
 	mmio_write_32(GPIO_OUTDT1, 0x00000000U);
-	mmio_write_32(GPIO_OUTDT2, 0x00000400U);
+	mmio_write_32(GPIO_OUTDT2, 0x00000000U);
 	mmio_write_32(GPIO_OUTDT3, 0x00000000U);
 	mmio_write_32(GPIO_OUTDT4, 0x00000000U);
 	mmio_write_32(GPIO_OUTDT5, 0x00000000U);
-	mmio_write_32(GPIO_OUTDT6, 0x00003800U);
-	mmio_write_32(GPIO_OUTDT7, 0x00000003U);
+	mmio_write_32(GPIO_OUTDT6, 0x00084800U);
+	mmio_write_32(GPIO_OUTDT7, 0x00000000U);
 
 	/* initialize general input/output switching */
-	mmio_write_32(GPIO_INOUTSEL0, 0x00000001U);
-	mmio_write_32(GPIO_INOUTSEL1, 0x00100B00U);
-	mmio_write_32(GPIO_INOUTSEL2, 0x00000418U);
-	mmio_write_32(GPIO_INOUTSEL3, 0x00002000U);
-	mmio_write_32(GPIO_INOUTSEL4, 0x00000040U);
+	mmio_write_32(GPIO_INOUTSEL0, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL1, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL2, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL3, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL4, 0x00000000U);
 #if (RCAR_GEN3_ULCB == 1)
-	mmio_write_32(GPIO_INOUTSEL5, 0x00000008U);
+	mmio_write_32(GPIO_INOUTSEL5, 0x00000000U);
 #else
-	mmio_write_32(GPIO_INOUTSEL5, 0x00000208U);
+	mmio_write_32(GPIO_INOUTSEL5, 0x00000000U);
 #endif
-	mmio_write_32(GPIO_INOUTSEL6, 0x00013F00U);
-	mmio_write_32(GPIO_INOUTSEL7, 0x00000003U);
+	mmio_write_32(GPIO_INOUTSEL6, 0x00084800U);
+	mmio_write_32(GPIO_INOUTSEL7, 0x00000000U);
 }
diff --git a/drivers/renesas/rzg/pfc/G2M/pfc_init_m3.c b/drivers/renesas/rzg/pfc/G2M/pfc_init_m3.c
index b87bced6a..432c4d3e7 100644
--- a/drivers/renesas/rzg/pfc/G2M/pfc_init_m3.c
+++ b/drivers/renesas/rzg/pfc/G2M/pfc_init_m3.c
@@ -970,7 +970,6 @@ void pfc_init_m3(void)
 	pfc_reg_write(PFC_GPSR6, GPSR6_AUDIO_CLKB_B
 		      | GPSR6_AUDIO_CLKA_A
 		      | GPSR6_SSI_WS6
-		      | GPSR6_SSI_SCK6
 		      | GPSR6_SSI_SDATA4
 		      | GPSR6_SSI_WS4
 		      | GPSR6_SSI_SCK4
@@ -1237,22 +1236,22 @@ void pfc_init_m3(void)
 	pfc_reg_write(PFC_DRVCTRL24, reg);
 
 	/* initialize LSI pin pull-up/down control */
-	pfc_reg_write(PFC_PUD0,0x00005FBFU);
-	pfc_reg_write(PFC_PUD1,0x00300EFEU);
-	pfc_reg_write(PFC_PUD2,0x330001E6U);
-	pfc_reg_write(PFC_PUD3,0x000002E0U);
-	pfc_reg_write(PFC_PUD4,0xFFFFFF00U);
-	pfc_reg_write(PFC_PUD5,0x7F5FFF87U);
-	pfc_reg_write(PFC_PUD6,0x00000055U);
+	pfc_reg_write(PFC_PUD0,0x00000000U);
+	pfc_reg_write(PFC_PUD1,0x00000000U);
+	pfc_reg_write(PFC_PUD2,0x00000000U);
+	pfc_reg_write(PFC_PUD3,0x00000000U);
+	pfc_reg_write(PFC_PUD4,0x00000000U);
+	pfc_reg_write(PFC_PUD5,0x00000000U);
+	pfc_reg_write(PFC_PUD6,0x00000000U);
 
 	/* initialize LSI pin pull-enable register */
-	pfc_reg_write(PFC_PUEN0,0x00000FFFU);
-	pfc_reg_write(PFC_PUEN1,0x00100234U);
-	pfc_reg_write(PFC_PUEN2,0x000004C4U);
-	pfc_reg_write(PFC_PUEN3,0x00000200U);
-	pfc_reg_write(PFC_PUEN4,0x3E000000U);
-	pfc_reg_write(PFC_PUEN5,0x1F000805U);
-	pfc_reg_write(PFC_PUEN6,0x00000006U);
+	pfc_reg_write(PFC_PUEN0,0x00000000U);
+	pfc_reg_write(PFC_PUEN1,0x00000000U);
+	pfc_reg_write(PFC_PUEN2,0x00000000U);
+	pfc_reg_write(PFC_PUEN3,0x00000000U);
+	pfc_reg_write(PFC_PUEN4,0x00000000U);
+	pfc_reg_write(PFC_PUEN5,0x00000000U);
+	pfc_reg_write(PFC_PUEN6,0x00000000U);
 
 	/* initialize positive/negative logic select */
 	mmio_write_32(GPIO_POSNEG0, 0x00000000U);
@@ -1275,27 +1274,27 @@ void pfc_init_m3(void)
 	mmio_write_32(GPIO_IOINTSEL7, 0x00000000U);
 
 	/* initialize general output register */
-	mmio_write_32(GPIO_OUTDT0, 0x00000001U);
+	mmio_write_32(GPIO_OUTDT0, 0x00000000U);
 	mmio_write_32(GPIO_OUTDT1, 0x00000000U);
-	mmio_write_32(GPIO_OUTDT2, 0x00000400U);
+	mmio_write_32(GPIO_OUTDT2, 0x00000000U);
 	mmio_write_32(GPIO_OUTDT3, 0x00000000U);
 	mmio_write_32(GPIO_OUTDT4, 0x00000000U);
 	mmio_write_32(GPIO_OUTDT5, 0x00000000U);
-	mmio_write_32(GPIO_OUTDT6, 0x00003800U);
-	mmio_write_32(GPIO_OUTDT7, 0x00000003U);
+	mmio_write_32(GPIO_OUTDT6, 0x00084800U);
+	mmio_write_32(GPIO_OUTDT7, 0x00000000U);
 
 	/* initialize general input/output switching */
-	mmio_write_32(GPIO_INOUTSEL0, 0x00000001U);
-	mmio_write_32(GPIO_INOUTSEL1, 0x00100B00U);
-	mmio_write_32(GPIO_INOUTSEL2, 0x00000418U);
-	mmio_write_32(GPIO_INOUTSEL3, 0x00002000U);
-	mmio_write_32(GPIO_INOUTSEL4, 0x00000040U);
+	mmio_write_32(GPIO_INOUTSEL0, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL1, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL2, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL3, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL4, 0x00000000U);
 #if (RCAR_GEN3_ULCB == 1)
-	mmio_write_32(GPIO_INOUTSEL5, 0x00000008U);
+	mmio_write_32(GPIO_INOUTSEL5, 0x00000000U);
 #else
-	mmio_write_32(GPIO_INOUTSEL5, 0x00000208U);
+	mmio_write_32(GPIO_INOUTSEL5, 0x00000000U);
 #endif
-	mmio_write_32(GPIO_INOUTSEL6, 0x00013F00U);
-	mmio_write_32(GPIO_INOUTSEL7, 0x00000003U);
+	mmio_write_32(GPIO_INOUTSEL6, 0x00084800U);
+	mmio_write_32(GPIO_INOUTSEL7, 0x00000000U);
 
 }
diff --git a/drivers/renesas/rzg/pfc/G2N/pfc_init_m3n.c b/drivers/renesas/rzg/pfc/G2N/pfc_init_m3n.c
index ebea3c20d..5e987555d 100644
--- a/drivers/renesas/rzg/pfc/G2N/pfc_init_m3n.c
+++ b/drivers/renesas/rzg/pfc/G2N/pfc_init_m3n.c
@@ -884,7 +884,6 @@ void pfc_init_m3n(void)
 	pfc_reg_write(PFC_GPSR6, GPSR6_AUDIO_CLKB_B
 		      | GPSR6_AUDIO_CLKA_A
 		      | GPSR6_SSI_WS6
-		      | GPSR6_SSI_SCK6
 		      | GPSR6_SSI_SDATA4
 		      | GPSR6_SSI_WS4
 		      | GPSR6_SSI_SCK4
@@ -1151,22 +1150,22 @@ void pfc_init_m3n(void)
 	pfc_reg_write(PFC_DRVCTRL24, reg);
 
 	/* initialize LSI pin pull-up/down control */
-	pfc_reg_write(PFC_PUD0,0x00005FBFU);
-	pfc_reg_write(PFC_PUD1,0x00300EFEU);
-	pfc_reg_write(PFC_PUD2,0x330001E6U);
-	pfc_reg_write(PFC_PUD3,0x000002E0U);
-	pfc_reg_write(PFC_PUD4,0xFFFFFF00U);
-	pfc_reg_write(PFC_PUD5,0x7F5FFF87U);
-	pfc_reg_write(PFC_PUD6,0x00000055U);
+	pfc_reg_write(PFC_PUD0,0x00000000U);
+	pfc_reg_write(PFC_PUD1,0x00000000U);
+	pfc_reg_write(PFC_PUD2,0x00000000U);
+	pfc_reg_write(PFC_PUD3,0x00000000U);
+	pfc_reg_write(PFC_PUD4,0x00000000U);
+	pfc_reg_write(PFC_PUD5,0x00000000U);
+	pfc_reg_write(PFC_PUD6,0x00000000U);
 
 	/* initialize LSI pin pull-enable register */
-	pfc_reg_write(PFC_PUEN0,0x00000FFFU);
-	pfc_reg_write(PFC_PUEN1,0x00100234U);
-	pfc_reg_write(PFC_PUEN2,0x000004C4U);
-	pfc_reg_write(PFC_PUEN3,0x00000200U);
-	pfc_reg_write(PFC_PUEN4,0x3E000000U);
-	pfc_reg_write(PFC_PUEN5,0x1F000805U);
-	pfc_reg_write(PFC_PUEN6,0x00000006U);
+	pfc_reg_write(PFC_PUEN0,0x00000000U);
+	pfc_reg_write(PFC_PUEN1,0x00000000U);
+	pfc_reg_write(PFC_PUEN2,0x00000000U);
+	pfc_reg_write(PFC_PUEN3,0x00000000U);
+	pfc_reg_write(PFC_PUEN4,0x00000000U);
+	pfc_reg_write(PFC_PUEN5,0x00000000U);
+	pfc_reg_write(PFC_PUEN6,0x00000000U);
 
 	/* initialize positive/negative logic select */
 	mmio_write_32(GPIO_POSNEG0, 0x00000000U);
@@ -1189,26 +1188,26 @@ void pfc_init_m3n(void)
 	mmio_write_32(GPIO_IOINTSEL7, 0x00000000U);
 
 	/* initialize general output register */
-	mmio_write_32(GPIO_OUTDT0, 0x00000001U);
+	mmio_write_32(GPIO_OUTDT0, 0x00000000U);
 	mmio_write_32(GPIO_OUTDT1, 0x00000000U);
-	mmio_write_32(GPIO_OUTDT2, 0x00000400U);
+	mmio_write_32(GPIO_OUTDT2, 0x00000000U);
 	mmio_write_32(GPIO_OUTDT3, 0x00000000U);
 	mmio_write_32(GPIO_OUTDT4, 0x00000000U);
 	mmio_write_32(GPIO_OUTDT5, 0x00000000U);
-	mmio_write_32(GPIO_OUTDT6, 0x00003800U);
-	mmio_write_32(GPIO_OUTDT7, 0x00000003U);
+	mmio_write_32(GPIO_OUTDT6, 0x00084800U);
+	mmio_write_32(GPIO_OUTDT7, 0x00000000U);
 
 	/* initialize general input/output switching */
-	mmio_write_32(GPIO_INOUTSEL0, 0x00000001U);
-	mmio_write_32(GPIO_INOUTSEL1, 0x00100B00U);
-	mmio_write_32(GPIO_INOUTSEL2, 0x00000418U);
-	mmio_write_32(GPIO_INOUTSEL3, 0x00002000U);
-	mmio_write_32(GPIO_INOUTSEL4, 0x00000040U);
+	mmio_write_32(GPIO_INOUTSEL0, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL1, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL2, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL3, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL4, 0x00000000U);
 #if (RCAR_GEN3_ULCB == 1)
-	mmio_write_32(GPIO_INOUTSEL5, 0x00000008U);
+	mmio_write_32(GPIO_INOUTSEL5, 0x00000000U);
 #else
-	mmio_write_32(GPIO_INOUTSEL5, 0x00000208U);
+	mmio_write_32(GPIO_INOUTSEL5, 0x00000000U);
 #endif
-	mmio_write_32(GPIO_INOUTSEL6, 0x00013F00U);
-	mmio_write_32(GPIO_INOUTSEL7, 0x00000003U);
+	mmio_write_32(GPIO_INOUTSEL6, 0x00084800U);
+	mmio_write_32(GPIO_INOUTSEL7, 0x00000000U);
 }
-- 
2.25.1

